scripts/sanitycheck

```diff
@@ -1918,20 +1918,21 @@ class TestInstance:
         # will pass this overlay to kconfig.py *twice* and kconfig.cmake
         # will silently give that second time precedence over any
         # --extra-args=CONFIG_*
-        subdir = os.path.join(self.outdir, "sanitycheck")
-        os.makedirs(subdir, exist_ok=True)
-        file = os.path.join(subdir, "testcase_extra.conf")
-        with open(file, "w") as f:
-            content = ""
+        if len(self.test.extra_configs) > 0 or options.coverage:
+            subdir = os.path.join(self.outdir, "sanitycheck")
+            os.makedirs(subdir, exist_ok=True)
+            file = os.path.join(subdir, "testcase_extra.conf")
+            with open(file, "w") as f:
+                content = ""
 
-            if len(self.test.extra_configs) > 0:
-                content = "\n".join(self.test.extra_configs)
+                if len(self.test.extra_configs) > 0:
```

Comment:
Ditto here.

(Just noticed it was like that in the old code too, but might as well.)